backend/src/server/api/v1beta/sharepoint.rs

```diff
@@ -0,0 +1,496 @@
+    ),
+    security(
+        ("bearer_auth" = [])
+    )
+)]
+pub async fn get_drive_item_children(
+    State(app_state): State<AppState>,
+    Extension(me_user): Extension<MeProfile>,
+    Path((drive_id, item_id)): Path<(String, String)>,
+) -> Result<Json<DriveItemsResponse>, StatusCode> {
+    check_sharepoint_enabled(&app_state)?;
+    let access_token = get_access_token(&me_user)?;
+    let client = create_graph_client(access_token);
+
+    let response = client
+        .drive(&drive_id)
+        .item(&item_id)
+        .list_children()
+        .send()
+        .await
+        .map_err(|e| {
+            tracing::error!("Failed to get folder children: {:?}", e);
+            StatusCode::INTERNAL_SERVER_ERROR
+        })?;
+
+    let items_json: serde_json::Value = response.json().await.map_err(|e| {
+        tracing::error!("Failed to parse folder children response: {:?}", e);
+        StatusCode::INTERNAL_SERVER_ERROR
+    })?;
+
+    let items = items_json
+        .get("value")
+        .and_then(|v| v.as_array())
+        .map(|arr| arr.iter().filter_map(parse_drive_item).collect())
+        .unwrap_or_default();
+
+    Ok(Json(DriveItemsResponse { items }))
+}
+
+/// Request to link a Sharepoint file to a chat
+#[derive(Debug, serde::Deserialize, ToSchema)]
+pub struct LinkSharepointFileRequest {
+    /// The ID of the chat to link the file to
+    pub chat_id: String,
+    /// The ID of the drive containing the file
+    pub drive_id: String,
+    /// The ID of the file item
+    pub item_id: String,
+}
+
+/// Response from linking a Sharepoint file
+#[derive(Debug, Serialize, ToSchema)]
+pub struct LinkSharepointFileResponse {
+    /// The ID of the created file upload record
+    pub file_id: String,
+    /// The filename of the linked file
+    pub filename: String,
+}
+
+/// Link a Sharepoint file to a chat.
+///
+/// This creates a file upload record that references the Sharepoint file,
+/// allowing it to be used in chat messages.
+#[utoipa::path(
+    post,
+    path = "/integrations/sharepoint/link-file",
+    tag = "sharepoint",
+    request_body = LinkSharepointFileRequest,
+    responses(
+        (status = OK, body = LinkSharepointFileResponse, description = "File linked successfully"),
+        (status = BAD_REQUEST, description = "Invalid request"),
+        (status = UNAUTHORIZED, description = "No access token available"),
+        (status = NOT_FOUND, description = "File not found or Sharepoint integration is not enabled"),
+        (status = INTERNAL_SERVER_ERROR, description = "Failed to link file")
+    ),
+    security(
+        ("bearer_auth" = [])
+    )
+)]
+pub async fn link_sharepoint_file(
+    State(app_state): State<AppState>,
+    Extension(me_user): Extension<MeProfile>,
+    Extension(policy): Extension<crate::policy::engine::PolicyEngine>,
+    Json(request): Json<LinkSharepointFileRequest>,
+) -> Result<Json<LinkSharepointFileResponse>, StatusCode> {
```

Comment:
This endpoint looks to special-purpose. Let's look into other endpoints that do the file-upload and allow the necessary data to be passed in there.